pkg/controller/s3/bucket/taggingConfig.go

```diff
@@ -61,13 +60,30 @@ func (in *TaggingConfigurationClient) Observe(ctx context.Context, bucket *v1bet
 		return Updated, nil
 	case config == nil && len(external.TagSet) != 0:
 		return NeedsDeletion, nil
-	case cmp.Equal(s3.SortS3TagSet(external.TagSet), s3.SortS3TagSet(GenerateTagging(config).TagSet)):
+	case IsS3TagsUpdated(s3.SortS3TagSet(external.TagSet), s3.SortS3TagSet(GenerateTagging(config).TagSet)):
```

Comment:
I think there is no need to sort the tags as you are now comparing the generated tags is a subset of external tags.

Reply:
good point